### PR TITLE
Add Lean Mode

### PR DESCRIPTION
**Pull request by 11Kilobytes**
I've added a mode for the Lean proof assistant (https://leanprover.github.io/). The layer adds additional layers to `dotspacemacs-configuration-layers` since the Lean mode for emacs requires packages that are owned by these layers.

### CODE REVIEW

File: layers/+lang/lean/README.org [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#+BEGIN_SRC elisp
+(setq-default lean-rootdir "/usr/local")
+#+END_SRC
+
+
+* Key Bindings
+
+| Key Binding                              | Description                                  |
+|------------------------------------------+----------------------------------------------|
+| @@html:<kbd>@@ SPC m L @@html:</kbd>@@   | Compile the current file/project.            |
```
Comment (person808) [inline layers/+lang/lean/README.org]: Use `~SPC m L~` for all keybindings.

File: layers/+lang/lean/packages.el [hunk 2]
```diff
@@ -0,0 +1,34 @@
+(setq dotspacemacs-additional-packages
+      (append '(dash dash-functional f s
+                     fill-column-indicator
+                     mmm-mode)))
```
Comment (person808) [inline layers/+lang/lean/packages.el]: Add these packages to `lean-packages`. Never touch `dotspacemacs-additional-packages` in a layer.

Comment (11Kilobytes) [inline layers/+lang/lean/packages.el]: Where should the required call to `setq` go, in `packages.el`?

Comment (person808) [inline layers/+lang/lean/packages.el]:
```elisp
(setq lean-packages '((lean-mode :location (recipe
                                            :fetcher github
                                            :repo leanprover/lean
                                            :files ("src/emacs/*.el")))
                                    fill-column-indicator
                                    mmm-mode))

Comment (11Kilobytes) [inline layers/+lang/lean/packages.el]: Thank you for helping me @person808, I'm almost done. However, I think I have a valid reason for adding things in `dotspacemacs-additional-packages`. The packages are "utility libraries," such as `s` which is a string manipulation library. For this reason, I don't think any package in particular should "own" these sorts of libraries.

**Comment by 11Kilobytes** [comment]
I messed up my repository while rebasing, so I'll just use multiple commits.

**Pull request closed without merging**